packages/cli/src/commands/deploy/helper.ts

```diff
@@ -414,3 +418,37 @@ export const createFieldPlugin = async (
     )
   }
 }
+
+// Publish Logic
+
+type CheckPublish = (params: {
+  publish: undefined | boolean
+  skipPrompts: undefined | boolean
+}) => Promise<boolean>
+
+export const checkPublish: CheckPublish = async ({ publish, skipPrompts }) => {
+  if (skipPrompts === true) {
+    //NOTE: Publish true is the default value
+    return publish || true
+  }
+
+  if (publish !== undefined) {
+    return publish
```

Comment:
```suggestion
    return !!publish
```

Just to be sure that it will return always a boolean even if `publish` is something else rather than `undefined` or `boolean`. Wdyt?

But maybe `commander` handles the parse automatically and it would never happen anyway 🤔

Reply:
Hey @demetriusfeijoo , great point. Indeed Commander does not allow anything else than boolean or undefined. When you try to specify a value like `--no-publish=value` it returns an error, so we should be save here. Thank you for pointing it out.

Reply:
We can use it `Boolean(expr)` instead of `!!expr` :)

Reply:
We can do it, but commander makes sure that no other value can be passed than boolean in this case.